sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -119,7 +119,30 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
       qualifiedTableName.database, qualifiedTableName.name)
 
     if (DDLUtils.isDatasourceTable(table)) {
-      val dataSourceTable = cachedDataSourceTables(qualifiedTableName)
+      val dataSourceTable =
+        cachedDataSourceTables(qualifiedTableName) match {
+          case l @ LogicalRelation(relation: HadoopFsRelation, _, _) =>
+            // Ignore the scheme difference when comparing the paths
+            val isSamePath =
+              table.storage.locationUri.isDefined && relation.location.rootPaths.size == 1 &&
+                table.storage.locationUri.get == relation.location.rootPaths.head.toUri.getPath
```

Comment:
Let me do more checks to see whether `rootPaths` could have more than one path when the table is partitioned.